documentation/decisions/011_unstyled_button.md

```diff
@@ -0,0 +1,99 @@
+---
+status: Approved
+date: 2023-04-25
+deciders: stephenjwatkins, OskiTheCoder, ldewald
+---
+
+# Support UnstyledButton component
+
+## Context and Problem Statement
+
+Easy UI needs to support a button that does the heavy lifting with regards to behavior and accessibility while not being opinionated with styling and the content it renders.
+
+## Decision Drivers
+
+- Non-disruptive to existing components
+- Accessibility
+- Easy to maintain
+- Easy to use
+- Reusable across multiple components
+
+## Options Considered
+
+- Create an `UnstyledButton` component that is internal to Easy UI.
+- Extend existing `IconButton` to support `UnstyledButton` functionality.
+
+## Decision Outcome
+
+After consideration, we have decided to move forward with creating an `UnstyledButton` component that is internal to Easy UI. Extending the `IconButton` would require additional thought into how to maintain a minimal and sensible API with additional complexity; also, we would likely be limited in the kind of content the variant would be able to support. With an internal `UnstyledButton` component, we don't have to disrupt any existing components and we can configure it to suit our needs without needing to consider outside use. To that point, since we wouldn't need to consider outside consumption, it should be slightly easier to maintain and handle breaking changes.
+
+## Design
+
+Similar to the `Button`, `DropdownButton`, and `IconButton` components, the `UnstyledButton` component will make use of React Aria's `useButton` hook to manage behavior and accessibility concerns.
+
+### API
+
+```ts
+export type UnstyledButtonProps = AriaButtonProps<"button"> & {
+  /** Content to render */
+  children?: ReactNode;
+  /* Classname to apply styles to button */
+  className?: string;
+  /**
+   * Disables button
+   * @default false
+   */
+  isDisabled?: boolean;
+};
+```
+
+### Anatomy
+
+```tsx
+import { useButton } from "react-aria";
+import { classNames } from "../utilities/css";
+
+export function UnstyledButton(props: UnstyledButtonProps) {
+  const { className, isDisabled = false, children = "Button" } = props;
+
+  const ref = useRef(null);
+  const { buttonProps } = useButton(props, ref);
+
+  return (
+    <button
+      disabled={isDisabled}
+      ref={ref}
+      className={classNames(className)}
+      {...buttonProps}
+    >
```

Comment:
**thought:** we'll want to ensure that all passed in props are spread on the component. i found that with Aria the `buttonProps` returned are only the ones relevant to the Aria library, and it removes some props that are still important to being passed into the element. if you see my `<Tooltip />` pull request, you'll notice that I spread the `...restProps` and `mergeProps()` onto the underlying element to ensure the appropriate props are passed in.

Reply:
this problem is slightly tricky. i ended up needing to make a helper in `Button/utilities` as part of the Tooltip PR to remove React Aria-specific props before spreading onto the `<button />` element. i'm open to better approaches.

i mentioned it in the Tooltip PR, but the problem is that the Aria `useButton()` hook swallows some props that need to get passed to the `<button />` element. most notably, the `onPointerEvent` and `onPointerLeave` props provided by the Tooltip trigger handler. essentially that means we have to ensure those props get spread to `<button />` independent of the `buttonProps` returned by the `useButton()` hook.

Reply:
interesting, your comment makes sense and i went ahead and included the utility function. ill take a closer look on the `Tooltip` PR